File: proton-sdk/incubating/client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/telemetry/VerificationErrorEvent.kt
package me.proton.drive.sdk.telemetry

data class VerificationErrorEvent(
    val volumeType: VolumeType,
    val field: EncryptedField,
    val fromBefore2024: Boolean,
    val addressMatchingDefaultShare: Boolean,
    val error: String?,
    val uid: String,
)

File: proton-sdk/client/cs/src/Proton.Drive.Sdk/Telemetry/VerificationErrorEvent.cs
using Proton.Drive.Sdk.Nodes;
using Proton.Sdk.Telemetry;

namespace Proton.Drive.Sdk.Telemetry;

public sealed class VerificationErrorEvent : IMetricEvent
{
    public string Name => "verificationError";

    public required VolumeType VolumeType { get; set; }

    public required EncryptedField Field { get; set; }

    public bool? FromBefore2024 { get; set; }

    public bool? AddressMatchingDefaultShare { get; set; }

    public string? Error { get; set; }

    public required NodeUid Uid { get; set; }
}
